be1-go/hub/standard_hub/message_handling.go

```diff
@@ -280,6 +281,22 @@ func (h *Hub) handlePublish(socket socket.Socket, byteMessage []byte) (int, erro
 	signature := publish.Params.Message.Signature
 	messageID := publish.Params.Message.MessageID
 	data := publish.Params.Message.Data
+	dataBytes, ok := base64.StdEncoding.DecodeString(data)
+	if ok != nil {
+		return publish.ID, xerrors.Errorf("Data is not base 64 encoded")
+	}
+	publicKeySender, ok := base64.URLEncoding.DecodeString(publish.Params.Message.Sender)
+	if ok != nil {
+		h.log.Info().Msg("Sender is : " + publish.Params.Message.Sender)
+		return publish.ID, xerrors.Errorf("Public key is not base 64 encoded " + ok.Error())
+	}
+	signatureBytes, ok := base64.URLEncoding.DecodeString(signature)
+	if ok != nil {
+		return publish.ID, xerrors.Errorf("Signature is not base 64 encoded")
+	}
+	if !ed25519.Verify(publicKeySender, dataBytes, signatureBytes) {
```

Comment:
You should use schnorr.Verify() instead of ed25519.Verify() because we use kyber on the go backend

Reply:
Okay thank you didn't know this, I updated it